### PR TITLE
Apply Operator Pattern/Unit Testing GetNotificationByID

### PR DESCRIPTION
**Pull request by nkstanchev**
Addresses #1588.

Signed-off-by: Nikolay Stanchev <nstanchev@vmware.com>

### CODE REVIEW

File: internal/support/notifications/operators/notification/db.go [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+ *******************************************************************************/
+
+package notification
+
+import contract "github.com/edgexfoundry/go-mod-core-contracts/models"
+
+// NotificationLoader provides functionality for obtaining Notifications.
+type NotificationLoader interface {
+	GetNotifications() ([]contract.Notification, error)
+	GetNotificationById(id string) (contract.Notification, error)
```
Comment (tsconn23) [inline internal/support/notifications/operators/notification/db.go]: This is the only method in this interface that is being used in this PR. What are the others for?

File: internal/support/notifications/operators/notification/db.go [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+	GetNotificationsByStartEnd(start int64, end int64, limit int) ([]contract.Notification, error)
+	GetNotificationsByStart(start int64, limit int) ([]contract.Notification, error)
+	GetNotificationsByEnd(end int64, limit int) ([]contract.Notification, error)
+	GetNewNotifications(limit int) ([]contract.Notification, error)
+	GetNewNormalNotifications(limit int) ([]contract.Notification, error)
+}
+
+// NotificationDeleter deletes notifications.
+type NotificationDeleter interface {
+	DeleteNotificationById(id string) error
```
Comment (tsconn23) [inline internal/support/notifications/operators/notification/db.go]: This is the only method in this interface that is being used in this PR. What are the others for?

**Comment by tsconn23** [comment]
@nkstanchev The codecov.io report on this PR says our coverage drops by %2. That seems like a lot and I'm trying to determine if it's an incorrect result. Can you answer the questions above and rebase this branch? I'm hoping the rebase will either retrigger codecov.io or at least allow us updated stats on their site. FWIW when I use `go tool cover -html=coverage.out` the coverage percentages are all very good.

File: internal/support/notifications/operators/notification/db.go (hunk 2 above)
Comment (nkstanchev) [reply internal/support/notifications/operators/notification/db.go]: The idea behind it was to set them up for future use but I didn't accounted for test coverage stats. I will removed them.

File: internal/support/notifications/operators/notification/db.go (hunk 1 above)
Comment (nkstanchev) [reply internal/support/notifications/operators/notification/db.go]: The idea behind it was to set them up for future use but I didn't accounted for test coverage stats. I will removed them.

**Pull request merged**